es-app/src/Gamelist.cpp

```diff
@@ -48,7 +48,12 @@ FileData* findOrCreateFile(SystemData* system, const std::string& path, FileType
 			}
 
 			FileData* file = new FileData(type, path, system->getSystemEnvData(), system);
-			treeNode->addChild(file);
+
+			// skipping arcade assets from gamelist
+			if(!treeNode->isArcadeAsset())
```

Comment:
Here's your error. treeNode will be the parent folder in this case. You probably want to run this check against "file", which is the newly-created file.

That being said, I'm not fully comfortable with the idea of even creating a FileData object if we're going to discard it - I'd rather only create it if we're going to use it, otherwise make sure to return NULL - and see if we don't crash.

Reply:
@pjft First: thank you for your time and patience.

I just pushed the requested changes but not yet the `NULL` return (before double check with you).

If I return NULL, it'll not crash, but it'll produce this output:

> lvl0: 	Error finding/creating FileData for "/home/rael/RetroPie/roms/neogeo/neogeo.zip", skipping.`

Returning the `FileData`, no error output.

How should I proceed?

Reply:
I'd leave a definitive answer for others to chime in on, but in my perspective I'm not wholly bothered by that warning. The alternative is to create FileData objects that are dangling without a parent, take up memory, and never get explicitly removed. I'd imagine there'd be some edge case where that'd come and bite us back.

But if anyone else has any recommendation or preference, I'd happily take that.